xml/src/main/java/org/ehcache/xml/ConfigurationParser.java

```diff
@@ -392,16 +414,26 @@ public static DocumentBuilder documentBuilder(Schema schema) throws ParserConfig
   }
 
   public static Schema discoverSchema(Source ... fixedSources) throws SAXException, IOException {
-    ArrayList<Source> schemaSources = new ArrayList<>(asList(fixedSources));
+    Map<URI, Source> pluginSchemas = new HashMap<>();
     for (CacheManagerServiceConfigurationParser<?> p : servicesOfType(CacheManagerServiceConfigurationParser.class)) {
-      schemaSources.add(p.getXmlSchema());
+      if (!pluginSchemas.containsKey(p.getNamespace())) {
+        pluginSchemas.put(p.getNamespace(), p.getXmlSchema());
+      }
     }
     for (CacheServiceConfigurationParser<?> p : servicesOfType(CacheServiceConfigurationParser.class)) {
-      schemaSources.add(p.getXmlSchema());
+      if (!pluginSchemas.containsKey(p.getNamespace())) {
+        pluginSchemas.put(p.getNamespace(), p.getXmlSchema());
+      }
     }
     for (CacheResourceConfigurationParser p : servicesOfType(CacheResourceConfigurationParser.class)) {
-      schemaSources.add(p.getXmlSchema());
+      if (!pluginSchemas.containsKey(p.getNamespace())) {
+        pluginSchemas.put(p.getNamespace(), p.getXmlSchema());
+      }
     }
+
+    List<Source> schemaSources = new ArrayList<>(asList(fixedSources));
```

Comment:
Does there need to be a guard against a fixed source being the same as a discovered plugin source?

Reply:
It would be coding error in Ehcache... and it wouldn't be very easy to enforce since we don't know until they are parsed what the namespaces of the fixed sources are.